esp-hal-common/Cargo.toml

```diff
@@ -43,19 +43,21 @@ esp32c3_pac = { package = "esp32c3", git = "https://github.com/esp-rs/esp-pacs.g
 esp32s2_pac = { package = "esp32s2", git = "https://github.com/esp-rs/esp-pacs.git", branch = "with_source", optional = true }
 esp32s3_pac = { package = "esp32s3", git = "https://github.com/esp-rs/esp-pacs.git", branch = "with_source", optional = true }
 
+[patch.crates-io]
+xtensa-lx-rt = { git = "https://github.com/esp-rs/xtensa-lx-rt", rev = "70c56b1090d36742077283af2a83ca4bc1b4f338" }
+
 [features]
-esp32   = [  "esp32_pac/rt", "xtensa",   "dual_core", "xtensa-lx-rt/esp32",   "xtensa-lx/esp32",   "smartled"]
-esp32c3 = ["esp32c3_pac/rt", "risc_v", "single_core", "smartled"]
-esp32s2 = ["esp32s2_pac/rt", "xtensa", "single_core", "xtensa-lx-rt/esp32s2", "xtensa-lx/esp32s2", "smartled"]
-esp32s3 = ["esp32s3_pac/rt", "xtensa",   "dual_core", "xtensa-lx-rt/esp32s3", "xtensa-lx/esp32s3", "smartled"]
+default = ["vectored"]
+esp32   = ["esp32_pac/rt",   "multicore", "xtensa-lx-rt/esp32",   "xtensa-lx/esp32", "procmacros/rtc_slow", "smartled"]
+esp32c3 = ["esp32c3_pac/rt", "unicore", "riscv", "riscv-atomic-emulation-trap", "smartled"]
+esp32s2 = ["esp32s2_pac/rt", "unicore", "xtensa-lx-rt/esp32s2", "xtensa-lx/esp32s2", "procmacros/rtc_slow", "smartled"] # TODO support xtensa atomic emulation
+esp32s3 = ["esp32s3_pac/rt", "multicore", "xtensa-lx-rt/esp32s3", "xtensa-lx/esp32s3", "procmacros/rtc_slow", "smartled"]
+
 
-# Architecture (should not be enabled directly, but instead by a PAC's feature)
-risc_v = ["riscv", "riscv-atomic-emulation-trap"]
-xtensa = ["procmacros/rtc_slow"]
 
 # Core Count (should not be enabled directly, but instead by a PAC's feature)
-single_core = []
-dual_core   = []
+unicore = []
+multicore   = []
```

Comment:
Curious why you've chosen `unicore` here instead of `single_core`, sounds a bit unnatural to me. Not a huge deal but I would have expected `single_core` and `multi_core` to be suitable. Not going to block merging based on this, though, so feel free to tell me to kick rocks.

Reply:
Not a huge deal to me either, just what I had in my head at the time. The main thing for me was removing dual core from the name. Any preference, @bjoernQ?

Reply:
I like Jesse's suggestion - I constantly read `unicorn` instead of `unicore`